sql/signal_handler.cc

```diff
@@ -58,49 +58,70 @@ extern const char *optimizer_switch_names[];
 
 static inline void output_core_info()
 {
-  /* proc is optional on some BSDs so it can't hurt to look */
-#if defined(HAVE_READLINK) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#ifndef __WIN__
   char buff[PATH_MAX];
   ssize_t len;
   int fd;
+  if ((fd= open("/etc/os-release", O_RDONLY)) >= 0)
+  {
+    len= read(fd, buff, sizeof(buff));
+    close(fd);
+    /* Take first 3 lines of this file, enough for distro and version only */
+    char *endline= (char *) memchr(buff, '\n', len);
```

Comment:
There is code repeatiton,instead I would write like so:
```C
  len_orig= read(fd, buff, sizeof(buff));
  len= len_orig;
  for (size_t num_lines= 0; num_lines <= 2 & len>0; num_lines++, endline++)
  {
    if (num_lines == 0)
      endline= (char *) memchr(buff, '\n', len_orig);
    else
      endline= (char *) memchr(endline, '\n', len_orig);
    len= endline - buff;
  }
```

Reply:
Right track. I'll do something like this. Generally if you doing an `if condition` on a loop variable inside the loop, one of the two is incorrect. In this example initialising `endline =buff` before the loop simplifies away the `if` condition.